src/preprocessing/passes/daneshvar.cpp

```diff
@@ -0,0 +1,650 @@
+                                "u" + std::string(8 - numDigits(id), '0') + std::to_string(id);
+                            Node newBv = nodeManager->mkBoundVar(new_var_name, bv.getType());
+                            boundVar2node[bv] = newBv;
+                            normalized[bv] = newBv;
+                        }
+                        normalizedBoundVars.push_back(normalized[bv]);
+                    }
+
+                    // Replace the bound variables in the quantifier
+                    Node normalizedBoundVarList = nodeManager->mkNode(
+                        cvc5::internal::Kind::BOUND_VAR_LIST, normalizedBoundVars);
+                    normalized[bound_vars] = normalizedBoundVarList;
+                }
+
+                // Push unvisited children onto the stack
+                for (int i = current.getNumChildren() - 1; i >= 0; i--)
+                {
+                    Node child = current[i];
+                    if (visited.find(child) == visited.end())
+                    {
+                        stack.push(child);
+                    }
+                }
+
+                // For APPLY_UF nodes, push the operator
+                if (current.getKind() == cvc5::internal::Kind::APPLY_UF)
+                {
+                    Node op = current.getOperator();
+                    if (visited.find(op) == visited.end())
+                    {
+                        stack.push(op);
+                    }
+                }
+
+                // Leave 'it->second' as false; we'll process this node later
+            }
+        }
+        else if (!it->second)
+        {
+            // Second time seeing this node, process it after its children
+
+            // Prepare children for node creation
+            std::vector<Node> children;
+
+            if (current.getKind() == cvc5::internal::Kind::APPLY_UF)
+            {
+                // Handle operator for APPLY_UF nodes
+                auto opIt = normalized.find(current.getOperator());
+                Assert(opIt != normalized.end());
+                children.push_back(opIt->second);
+            }
+            else if (current.getMetaKind() == metakind::PARAMETERIZED)
+            {
+                // For parameterized nodes, include the operator
+                children.push_back(current.getOperator());
+            }
+
+            // Add normalized children
+            for (size_t i = 0; i < current.getNumChildren(); ++i)
+            {
+                Node child = current[i];
+
+                // For quantifiers, use the normalized bound variable list
+                if ((current.getKind() == cvc5::internal::Kind::FORALL ||
+                     current.getKind() == cvc5::internal::Kind::EXISTS) &&
+                    i == 0)
+                {
+                    // First child is the bound variable list
+                    children.push_back(normalized[child]);
+                    continue;
+                }
```

Comment:
Why is this needed? It doesn't do anything else than the code below.

Reply:
True. Removed unnecessary if.